### PR TITLE
Move Findings and Alerts action, request, response and models from alerting to common-utils

### PR DESCRIPTION
**Pull request by eirsep**
Move Findings and Alerts action, request, response and models from alerting to common-utils

Signed-off-by: Surya Sashank Nistala <snistala@amazon.com>

### CODE REVIEW

File: src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    val startTime: Instant,
+    val endTime: Instant? = null,
+    val lastNotificationTime: Instant? = null,
+    val acknowledgedTime: Instant? = null,
+    val errorMessage: String? = null,
+    val errorHistory: List<AlertError>,
+    val severity: String,
+    val actionExecutionResults: List<ActionExecutionResult>,
+    val aggregationResultBucket: AggregationResultBucket? = null
+) : Writeable, ToXContent {
```
Comment (getsaurabh02) [inline src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:42]: Is there a room to segregate the PoJo for Alert model required for response vs one required for data persistence and transport.

**Review by getsaurabh02 — APPROVED** [review]

**Comment by getsaurabh02** [comment]
We need to address the build failure

**Review by getsaurabh02 — DISMISSED** [review]

**Comment by sbcd90** [comment]
@eirsep @getsaurabh02 we see that `test coverage` for `common-utils` reduced from `75.91%` to `70.02%`. we need to move unit tests from `alerting` also along with moving the `model classes`.

e.g.  https://github.com/opensearch-project/alerting/blob/2.x/alerting/src/test/kotlin/org/opensearch/alerting/model/AlertTests.kt
https://github.com/opensearch-project/alerting/blob/2.x/alerting/src/test/kotlin/org/opensearch/alerting/model/XContentTests.kt
https://github.com/opensearch-project/alerting/blob/2.x/alerting/src/test/kotlin/org/opensearch/alerting/action/GetAlertsRequestTests.kt
https://github.com/opensearch-project/alerting/blob/2.x/alerting/src/test/kotlin/org/opensearch/alerting/action/GetAlertsResponseTests.kt

**Review by sbcd90 — DISMISSED** [review]

**Comment by eirsep** [comment]
Copied over findings models, actions, request, response classes also

@sbcd90 @getsaurabh02 plz review the latest commit too

**Review by getsaurabh02 — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**